pkg/config/cert_rotation_config.go

```diff
@@ -29,15 +33,136 @@ type CertRotationConfig struct {
+
+	// TODO: This is intended to be temporary, and will eventually be retrieved from a persistent external datastore
+	// https://github.com/abcxyz/jvs/issues/17
+	// KeyName format: `projects/*/locations/*/keyRings/*/cryptoKeys/*`
+	// https://pkg.go.dev/google.golang.org/genproto/googleapis/cloud/kms/v1#CryptoKey
+	KeyNames []string `yaml:"key_names,omitempty" env:"KEY_NAMES,overwrite"`
 }
 
 // Validate checks if the config is valid.
-func (c *CertRotationConfig) Validate() error {
-	var merr *multierror.Error
+func (cfg *CertRotationConfig) Validate() error {
+	var err *multierror.Error
+
+	if cfg.KeyTTL <= 0 {
+		err = multierror.Append(err, fmt.Errorf("key ttl must be a positive duration, got %q", cfg.KeyTTL))
+	}
+
+	if cfg.GracePeriod <= 0 {
+		err = multierror.Append(err, fmt.Errorf("grace period must be a positive duration, got %q", cfg.GracePeriod))
+	}
+
+	if cfg.DisabledPeriod <= 0 {
+		err = multierror.Append(err, fmt.Errorf("disabled period must be a positive duration, got %q", cfg.DisabledPeriod))
+	}
 
-	if err := c.CryptoConfig.Validate(); err != nil {
-		merr = multierror.Append(merr, err)
+	// Propagation delay must be positive but less than the grace period.
+	if cfg.PropagationDelay <= 0 {
+		err = multierror.Append(err, fmt.Errorf("propagation delay must be a positive duration, got %q", cfg.PropagationDelay))
+	}
+	if cfg.PropagationDelay > cfg.GracePeriod {
+		err = multierror.Append(err, fmt.Errorf("propagation delay %q must be less than grace period %q",
+			cfg.PropagationDelay, cfg.GracePeriod))
 	}
 
-	return merr.ErrorOrNil()
+	return err.ErrorOrNil()
+}
+
+// RotationAge gets the duration after a key has been created that a new key should be created.
+func (cfg *CertRotationConfig) RotationAge() time.Duration {
+	return cfg.KeyTTL - cfg.GracePeriod
+}
+
+// DestroyAge gets the duration after a key has been created when it becomes a candidate to be destroyed.
+func (cfg *CertRotationConfig) DestroyAge() time.Duration {
+	return cfg.KeyTTL + cfg.DisabledPeriod
+}
+
+// ToFlags returns a [cli.FlagSet] that is bound to the config.
+func (cfg *CertRotationConfig) ToFlags(opts ...cli.Option) *cli.FlagSet {
+	set := cli.NewFlagSet(opts...)
+
+	// Command options
+	f := set.NewSection("COMMON SERVER OPTIONS")
+
+	f.StringVar(&cli.StringVar{
+		Name:   "project-id",
+		Target: &cfg.ProjectID,
+		EnvVar: "PROJECT_ID",
+		Usage:  `Google Cloud project ID.`,
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:    "dev-mode",
+		Target:  &cfg.DevMode,
+		EnvVar:  "DEV_MODE",
+		Default: false,
+		Usage:   "Set to true to enable more granular debugging in logs",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "port",
+		Target:  &cfg.Port,
+		EnvVar:  "PORT",
+		Default: "8080",
+		Usage:   `The port the server listens to.`,
+	})
+
+	f = set.NewSection("KEY ROTATION OPTIONS")
+
+	f.DurationVar(&cli.DurationVar{
+		Name:    "key-ttl",
+		Target:  &cfg.KeyTTL,
+		EnvVar:  "KEY_TTL",
```

Comment:
With the exception of port, dev, and project_id, should some of these envvars be prefixed?